### ISSUE TITLE
Documentation: Duplicate object description, use :noindex: for one of them

### ISSUE THREAD

**Issue by adamjstewart** [LLNL/spack#1641]
Does anyone know what this means or how to fix it?
```
/blues/gpfs/home/software/spack-0.9.1/lib/spack/spack/__init__.py:docstring of spack.patch:0: WARNING: duplicate object description of spack.patch, other instance in /blues/gpfs/home/software/spack-0.9.1/lib/spack/docs/spack.rst, use :noindex: for one of them
/blues/gpfs/home/software/spack-0.9.1/lib/spack/spack/__init__.py:docstring of spack.version:0: WARNING: duplicate object description of spack.version, other instance in /blues/gpfs/home/software/spack-0.9.1/lib/spack/docs/spack.rst, use :noindex: for one of them
/blues/gpfs/home/software/spack-0.9.1/lib/spack/spack/__init__.py:docstring of spack.variant:0: WARNING: duplicate object description of spack.variant, other instance in /blues/gpfs/home/software/spack-0.9.1/lib/spack/docs/spack.rst, use :noindex: for one of them
/blues/gpfs/home/software/spack-0.9.1/lib/spack/spack/__init__.py:docstring of spack.resource:0: WARNING: duplicate object description of spack.resource, other instance in /blues/gpfs/home/software/spack-0.9.1/lib/spack/docs/spack.rst, use :noindex: for one of them
```

**Comment by adamjstewart** [comment]
This is the last remaining bug to be solved before #1576 is ready to go.

**Comment by adamjstewart** [comment]
This bug is also described here: https://github.com/astropy/astropy/issues/317 Still trying to wrap my head around it.

**Comment by adamjstewart** [comment]
Fixed in #1576. Thanks to @tgamblin for the solution.
